src/storage-operators/src/s3_oneshot_sink.rs

```diff
@@ -188,38 +190,50 @@ impl CopyToS3Uploader {
         }
     }
 
-    /// Creates the uploader for the next file which starts the multi part upload.
-    async fn start_new_file(&mut self) -> Result<(), anyhow::Error> {
+    /// Creates the uploader for the next file and starts the multi part upload.
+    async fn start_new_file_upload(&mut self) -> Result<(), anyhow::Error> {
         self.flush().await?;
         assert!(self.current_file_uploader.is_none());
 
         self.file_index += 1;
+        let file_path = self.current_file_path();
+
+        let bucket = self.bucket.clone();
+        info!("starting upload: bucket {}, file {}", &bucket, &file_path);
+        let sdk_config = self
+            .sdk_config
+            .take()
+            .expect("sdk_config should always be present");
+        let max_file_size = self.max_file_size;
+        let handle = mz_ore::task::spawn(|| "s3_uploader::try_new", async move {
```

Comment:
the existence of these tasks need a comment (explaining that we don't want to io on timely threads)

Reply:
makes sense, will add

Reply:
updated